toolchain/check/context.h

```diff
@@ -222,6 +222,17 @@ class Context {
       std::optional<llvm::function_ref<auto()->DiagnosticBuilder>> diagnoser =
           std::nullopt) -> bool;
 
+  // Returns the type `type_id` as a complete type, or produces an incomplete
+  // type error and returns an error type. This is a convenience wrapper around
+  // TryToCompleteType.
+  auto AsCompleteType(
```

Comment:
Should we have TryAsCompleteType for consistency, or do you still want "To" because it's returning `bool` instead of an optional?

Reply:
Yes, I would like to keep the `To` because I'd expect `TryAsCompleteType` to return an `optional<TypeId>`.

Also, because the diagnoser is now mandatory for `AsCompleteType` (which seems right because we shouldn't form a `TypeId::Error` without diagnosing) they're slightly less similar, so giving them slightly less similar names seems a bit better.